src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+        {
+            return await GetMetricsAsync();
+        }
+
+        public async Task<EventHubTriggerMetrics> GetMetricsAsync()
+        {
+            EventHubTriggerMetrics metrics = new EventHubTriggerMetrics();
+            EventHubRuntimeInformation runtimeInfo = null;
+
+            try
+            {
+                runtimeInfo = await _client.GetRuntimeInformationAsync();
+            }
+            catch (NotSupportedException e)
+            {
+                _logger.LogWarning($"Event Hub Trigger does not support Notification Hubs. Error: {e.Message}");
+                return metrics;
+            }
+            catch (MessagingEntityNotFoundException)
+            {
+                _logger.LogWarning($"Event Hub '{_eventHubName}' was not found.");
+                return metrics;
+            }
+            catch (TimeoutException e)
+            {
+                _logger.LogWarning($"Encountered a timeout while checking Event Hub '{_eventHubName}'. Error: {e.Message}");
+                return metrics;
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"Encountered an exception while checking Event Hub '{_eventHubName}'. Error: {e.Message}");
+                return metrics;
+            }
+
+            // Get the PartitionRuntimeInformation for all partitions
+            var tasks = new Task<EventHubPartitionRuntimeInformation>[runtimeInfo.PartitionCount];
+
+            for (int i = 0; i < runtimeInfo.PartitionCount; i++)
+            {
+                tasks[i] = _client.GetPartitionRuntimeInformationAsync(i.ToString());
+            }
+
+            await Task.WhenAll(tasks);
+
+            return await CreateTriggerMetrics(tasks.Select(t => t.Result).ToList());
+        }
+
+        internal async Task<EventHubTriggerMetrics> CreateTriggerMetrics(List<EventHubPartitionRuntimeInformation> partitionRuntimeInfo, bool alwaysLog = false)
+        {
+            long totalUnprocessedEventCount = 0;
+            bool logPartitionInfo = alwaysLog ? true : DateTime.UtcNow >= _nextPartitionLogTime;
+            bool logPartitionWarning = alwaysLog ? true : DateTime.UtcNow >= _nextPartitionWarningTime;
+
+            // For each partition, get the last enqueued sequence number.
+            // If the last enqueued sequence number does not equal the SequenceNumber from the lease info in storage,
+            // accumulate new event counts across partitions to derive total new event counts.
+            List<string> partitionErrors = new List<string>();
+            for (int i = 0; i < partitionRuntimeInfo.Count; i++)
+            {
+                long partitionUnprocessedEventCount = 0;
+
+                Tuple<BlobPartitionLease, string> partitionLeaseFile = await GetPartitionLeaseFileAsync(i);
+                BlobPartitionLease partitionLeaseInfo = partitionLeaseFile.Item1;
+                string error = partitionLeaseFile.Item2;

```

Comment:
This will be an error message?

Reply:
Yes -- I changed to errorMsg to make that more clear from the caller side.